lib/barlix/ean13.ex

```diff
@@ -0,0 +1,171 @@
+defmodule Barlix.EAN13 do
+  require Integer
+  use Bitwise, only_operators: true
+  import Integer, only: [mod: 2]
+
+  @moduledoc """
+  Implements [EAN13](https://en.wikipedia.org/wiki/International_Article_Number).
+  """
+
+  @doc """
+  Encodes the given value using EAN13. The given code is validated first.
+
+  ## Examples:
+
+    iex> Barlix.EAN13.encode("5449000096241")
+    {:ok, {:D1, [
+    1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 1
+    ]}}
+
+
+    iex> Barlix.EAN13.encode("5901234123450")
+    {:error, "validation failed: expected checksum digit 7 but received 0"}
+
+  """
+  @spec encode(String.t()) :: {:error, String.t()} | {:ok, Barlix.code()}
+  def encode(value) when is_binary(value) and byte_size(value) == 13 do
```

Comment:
btw, do we need the guard here? same with encode!. Just leave it to validate?

Reply:
We probably don't, right. I like them nowaways because they also kind of document intent. It leads to »no function clause matching« if called with something different.

Should I remove them?

Reply:
I am ok with is_binary which asserts on the type and should never happen on production, but byte_size might make things complicated, like let's say I ask the input from user, then returning `{:error, message}` is desirable.